### PR TITLE
Two small improvements

### PR DESCRIPTION
**Pull request by ensonic**
* mentions dependencies in the docs
* make the namespace attribute on k8s_object optional

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: README.md [hunk 1]
```diff
@@ -64,6 +64,11 @@ for setting up authentication:
 gcloud container clusters get-credentials <CLUSTER NAME>
 ```
 
+## Dependencies
+
+The rules will require the `kubectl` and `gcloud` tools when executing the `run`
+action from bazel.
```
Comment (mattmoor) [inline README.md]: Technically `gcloud` isn't required unless you are using GCP.

**Comment by mattmoor** [comment]
Test this please

**Comment by mattmoor** [comment]
Thanks for the PR @ensonic 

One small nit, but otherwise looks good.  I'll merge it into a branch and then open a PR to merge things from there so our integration testing will run since Travis doesn't provide secrets unless the PR is from a branch on the main repo.

**Comment by ensonic** [comment]
I'll update the doc patch. I run the tests, but this is probably not covered. I'll take another look and see if I can add a test for it.

**Comment by mattmoor** [comment]
Sorry, that comment was for Bazel CI :)

**Comment by ensonic** [comment]
Ping?

**Comment by mattmoor** [comment]
Sorry I didn't realize you'd pushed more commits.  If you can rebase I'll merge this into a branch so we can run CI, and then into master.

**Comment by ensonic** [comment]
Rebased.

**Comment by mattmoor** [comment]
Test this please

(for Bazel CI)

**Comment by mattmoor** [comment]
Hmm, something is broken.  I'm seeing: 
```
error: arguments in resource/name form may not have more than one slash
```

File: k8s/delete.sh.tpl [hunk 2]
```diff
@@ -23,5 +23,4 @@ function guess_runfiles() {
 
 RUNFILES="${PYTHON_RUNFILES:-$(guess_runfiles)}"
 
-kubectl --cluster="%{cluster}" --namespace="%{namespace}" delete \
-    -f %{unresolved}
+kubectl --cluster="%{cluster}" %{namespace_arg} delete k8s/...-f %{unresolved}
```
Comment (mattmoor) [inline k8s/delete.sh.tpl]: `k8s/...` looks like the problem

**Comment by ensonic** [comment]
Sorry about it. No idea how this ended up there. Thanks for testing all ops too. Fixed and rebased.

**Comment by ensonic** [comment]
Oh, you merged already. Thanks for cleaning it up.

**Comment by mattmoor** [comment]
Thanks for the PR :)

**Pull request merged**